### PR TITLE
ALL：fixing static analysis warnings

### PR DESCRIPTION
**Pull request by felixYyu**
this PR fixed these grouping parentheses are unnecessary. cc @rdblue @RussellSpitzer 
```
iceberg\core\src\main\java\org\apache\iceberg\BaseFileScanTask.java:224: [UnnecessaryParentheses] These grouping parentheses are unnecessary; it is unlikely the code will be misinterpreted without them
          (this.file().equals(other.file())) &&
          ^
    (see https://errorprone.info/bugpattern/UnnecessaryParentheses)
  Did you mean 'this.file().equals(other.file()) &&'?
```

### CODE REVIEW

**Comment by RussellSpitzer** [comment]
Can you combine some of these style prs and perhaps label them as fixing static analysis warnings

File: core/src/main/java/org/apache/iceberg/BaseFileScanTask.java [hunk 1]
```diff
@@ -220,12 +220,13 @@ public Expression residual() {
...
-      return (other != null) &&
-          (this.file().equals(other.file())) &&
-          (this.offset + this.len == other.offset);
+      return other != null &&
+          this.file().equals(other.file()) &&
+          this.offset + this.len == other.offset;
     }
   }
 
+  @SuppressWarnings("MixedMutabilityReturnType")
```
Comment (rdblue) [inline core/src/main/java/org/apache/iceberg/BaseFileScanTask.java:229]: This seems unrelated to the parentheses change. What is the purpose of this?

**Comment by felixYyu** [comment]
Can you combine some of these style prs and perhaps label them as fixing static analysis warnings
is a good idea, I try do it. @RussellSpitzer

File: core/src/main/java/org/apache/iceberg/BaseFileScanTask.java (hunk 1 above)
Comment (felixYyu) [reply core/src/main/java/org/apache/iceberg/BaseFileScanTask.java:229]: I will fixing all static analysis warnings with this PR.

**Comment by felixYyu** [comment]
updated list:
```
Object                    Warnings
------------------------------------------------------------------------------------
BaseFileScanTask.java               UnnecessaryParentheses/MixedMutabilityReturnType
BaseFilesTable.java                 SummaryFragment
FlinkCatalog.java                   MixedMutabilityReturnType
FlinkSplitPlanner.java              LambdaMethodReference
SnapshotRef.java                    ObjectEqualsForPrimitives
CharSequenceSet.java                ObjectsHashCodeUnnecessaryVarargs
DynClasses.java                     ImplicitPublicBuilderConstructor
DynFields.java                      ImplicitPublicBuilderConstructor
MappedFields.java                   ObjectsHashCodeUnnecessaryVarargs
MetricsModes.java                   ObjectsHashCodeUnnecessaryVarargs
BuildAvroProjection.java            UnnecessarilyQualified
SnapshotSummary.java                ImplicitPublicBuilderConstructor
StructLikeSet.java                  ObjectsHashCodeUnnecessaryVarargs
JdbcUtil.java                       ProtectedMembersInFinalClass
RemoveSnapshots.java                JavaUtilDate
LoadTableResponse.java              MissingSummary
ScanSummary.java                    JavaUtilDate
ParquetMetricsRowGroupFilter.java   UnusedMethod
IcebergSplit.java                   MutablePublicArray
TestIcebergInputFormats.java        MutablePublicArray
------------------------------------------------------------------------------------ 
```

File: api/src/main/java/org/apache/iceberg/SnapshotRef.java [hunk 2]
```diff
@@ -86,7 +86,7 @@ public boolean equals(Object other) {
     }
 
     SnapshotRef ref = (SnapshotRef) other;
-    return Objects.equals(ref.snapshotId(), snapshotId) &&
+    return ref.snapshotId() == snapshotId &&
```
Comment (RussellSpitzer) [inline api/src/main/java/org/apache/iceberg/SnapshotRef.java:89]: I think it's fine if we disable the static analysis rule for this, just because I like it when all the comparisons look the same but if we want to follow this rule that's also ok.

File: api/src/main/java/org/apache/iceberg/util/CharSequenceSet.java [hunk 3]
```diff
@@ -172,7 +172,7 @@ public boolean equals(Object o) {
 
   @Override
   public int hashCode() {
-    return Objects.hash(wrapperSet);
+    return Objects.hashCode(wrapperSet);
```
Comment (RussellSpitzer) [inline api/src/main/java/org/apache/iceberg/util/CharSequenceSet.java:175]: I'm not very comfortable changing this because it's a public api. While I don't think we actually serialize this anywhere, if someone did this would probably cause an issue. That said if as a community we feel confident this isn't going to be breaking we can change it.

Objects.hash will hash the hash of wrapperSet 
Objects.hashCode will hash wrapperSet

File: core/src/main/java/org/apache/iceberg/rest/responses/LoadTableResponse.java [hunk 4]
```diff
@@ -27,7 +27,7 @@
 import org.apache.iceberg.relocated.com.google.common.collect.Maps;
 
 /**
- *
+ * Table metadata result when loading a table
```
Comment (RussellSpitzer) [inline core/src/main/java/org/apache/iceberg/rest/responses/LoadTableResponse.java:30]: I believe @rdblue is in the middle of following up here as well, I think in this case I would just drop the empty javadoc in anticipation of a commit which adds a more specific doc.

File: parquet/src/main/java/org/apache/iceberg/parquet/ParquetMetricsRowGroupFilter.java [hunk 5]
```diff
@@ -565,6 +565,7 @@ private static boolean mayContainNull(Statistics statistics) {
     return !statistics.isNumNullsSet() || statistics.getNumNulls() > 0;
   }
 
+  @SuppressWarnings("UnusedMethod")
```
Comment (RussellSpitzer) [inline parquet/src/main/java/org/apache/iceberg/parquet/ParquetMetricsRowGroupFilter.java:568]: If this is really unused shouldn't we be removing it? Is it needed for something via reflection?

File: api/src/main/java/org/apache/iceberg/SnapshotRef.java (hunk 2 above)
Comment (felixYyu) [reply api/src/main/java/org/apache/iceberg/SnapshotRef.java:89]: revert

File: api/src/main/java/org/apache/iceberg/util/CharSequenceSet.java (hunk 3 above)
Comment (felixYyu) [reply api/src/main/java/org/apache/iceberg/util/CharSequenceSet.java:175]: revert

File: core/src/main/java/org/apache/iceberg/MetricsModes.java [hunk 6]
```diff
@@ -134,7 +134,7 @@ public boolean equals(Object other) {
 
     @Override
     public int hashCode() {
-      return Objects.hash(length);
+      return Integer.hashCode(length);
```
[the remark this answers is not in the document]
Comment (felixYyu) [reply core/src/main/java/org/apache/iceberg/MetricsModes.java:137]: to maintain hash consistency, so revert

File: core/src/main/java/org/apache/iceberg/ScanSummary.java [hunk 7]
```diff
@@ -295,7 +295,7 @@ private PartitionMetrics updateFromFile(ContentFile<?> file, Long timestampMilli
     @Override
     public String toString() {
       String dataTimestamp = dataTimestampMillis != null ?
-          new Date(dataTimestampMillis).toString() : null;
+              Instant.ofEpochMilli(dataTimestampMillis).toString() : null;
```
[the remark this answers is not in the document]
Comment (felixYyu) [reply core/src/main/java/org/apache/iceberg/ScanSummary.java:298]: fixed

File: core/src/main/java/org/apache/iceberg/jdbc/JdbcUtil.java [hunk 8]
```diff
@@ -30,6 +30,7 @@
 import org.apache.iceberg.relocated.com.google.common.base.Splitter;
 import org.apache.iceberg.relocated.com.google.common.collect.Iterables;
 
+@SuppressWarnings("ProtectedMembersInFinalClass")
```
[the remark this answers is not in the document]
Comment (felixYyu) [reply core/src/main/java/org/apache/iceberg/jdbc/JdbcUtil.java:33]: because the JdbcUtil's vars are used in orther class, so i want to change protected members to public.

File: core/src/main/java/org/apache/iceberg/rest/responses/LoadTableResponse.java (hunk 4 above)
Comment (felixYyu) [reply core/src/main/java/org/apache/iceberg/rest/responses/LoadTableResponse.java:30]: fix via drop the empty javadoc

File: flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java [hunk 9]
```diff
@@ -486,6 +486,7 @@ private static PartitionSpec toPartitionSpec(List<String> partitionKeys, Schema
     return builder.build();
   }
 
+  @SuppressWarnings("MixedMutabilityReturnType")
```
[the remark this answers is not in the document]
Comment (felixYyu) [reply flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java:489]: fix using an immutable list builder

File: parquet/src/main/java/org/apache/iceberg/parquet/ParquetMetricsRowGroupFilter.java (hunk 5 above)
Comment (felixYyu) [reply parquet/src/main/java/org/apache/iceberg/parquet/ParquetMetricsRowGroupFilter.java:568]: not sure via reflection, so revert

**Comment by felixYyu** [comment]
changelist-v2

```
Object                                Warnings
------------------------------------------------------------------------------------
Tasks.java                          StreamToIterable/new Runnable() lambda
Avro.java                           UnnecessaryLambda
InputFilesDecryptor.java            StreamToIterable
ArrowReader.java                    StreamToIterable
------------------------------------------------------------------------------------ 
```

**Comment by felixYyu** [comment]
please review again，thanks @RussellSpitzer

File: arrow/src/main/java/org/apache/iceberg/arrow/vectorized/ArrowReader.java [hunk 10]
```diff
@@ -268,7 +268,7 @@ public void close() throws IOException {
           .map(entry -> EncryptedFiles.encryptedInput(io.newInputFile(entry.getKey()), entry.getValue()));
 
       // decrypt with the batch call to avoid multiple RPCs to a key server, if possible
-      Iterable<InputFile> decryptedFiles = encryptionManager.decrypt(encrypted::iterator);
+      Iterable<InputFile> decryptedFiles = encryptionManager.decrypt(encrypted.collect(Collectors.toList()));
```
Comment (RussellSpitzer) [inline arrow/src/main/java/org/apache/iceberg/arrow/vectorized/ArrowReader.java:271]: This will fully materialize the list so I'm not sure we want to change the approach here. Currently the only usages do single passes over the iterable (see the forEach below) ... It is probably fine to suppress here but I also don't think  we lose to much to much by materializing right here since we immediately put our results into a materialized map anyway.

I guess i'm leaning towards this being ok at the moment although I don't think it materially improves the code

File: core/src/main/java/org/apache/iceberg/util/Tasks.java [hunk 11]
```diff
@@ -570,7 +566,7 @@ public Integer next() {
   }
 
   public static <I> Builder<I> foreach(Stream<I> items) {
-    return new Builder<>(items::iterator);
+    return new Builder<>(items.collect(Collectors.toList()));
```
Comment (RussellSpitzer) [inline core/src/main/java/org/apache/iceberg/util/Tasks.java:569]: Our "iterator<I> items" already makes an iterable out of an iterator so I don't think we should avoid it here. This change also forces a materialization so we should avoid it.

File: flink/v1.12/flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java [hunk 12]
```diff
@@ -488,16 +489,18 @@ private static PartitionSpec toPartitionSpec(List<String> partitionKeys, Schema
...
+    ImmutableList.Builder<String> partitionKeysBuilder = ImmutableList.builder();
     for (PartitionField field : spec.fields()) {
       if (field.transform().isIdentity()) {
-        partitionKeys.add(icebergSchema.findColumnName(field.sourceId()));
+        partitionKeysBuilder.add(icebergSchema.findColumnName(field.sourceId()));
       } else {
         // Not created by Flink SQL.
         // For compatibility with iceberg tables, return empty.
         // TODO modify this after Flink support partition transform.
-        return Collections.emptyList();
```
Comment (RussellSpitzer) [inline flink/v1.12/flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java:498]: This could stay Collections.emptyList()

File: flink/v1.12/flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java [hunk 13]
```diff
@@ -488,16 +489,18 @@ private static PartitionSpec toPartitionSpec(List<String> partitionKeys, Schema
...
+        return partitionKeys;
       }
     }
+    partitionKeys = partitionKeysBuilder.build();
     return partitionKeys;
```
Comment (RussellSpitzer) [inline flink/v1.12/flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java:504]: This could return partitionKeysBuilder.build()

File: flink/v1.12/flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java [hunk 14]
```diff
@@ -488,16 +489,18 @@ private static PartitionSpec toPartitionSpec(List<String> partitionKeys, Schema
 
   private static List<String> toPartitionKeys(PartitionSpec spec, Schema icebergSchema) {
     List<String> partitionKeys = Lists.newArrayList();
+    ImmutableList.Builder<String> partitionKeysBuilder = ImmutableList.builder();
```
[the remark this answers is not in the document]
Comment (RussellSpitzer) [reply flink/v1.12/flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java:492]: Since we don't need `partitionKeys` it may be ok to just call this builder `partitionKeys`

File: flink/v1.13/flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java [hunk 15]
```diff
@@ -488,16 +489,18 @@ private static PartitionSpec toPartitionSpec(List<String> partitionKeys, Schema
 
   private static List<String> toPartitionKeys(PartitionSpec spec, Schema icebergSchema) {
     List<String> partitionKeys = Lists.newArrayList();
+    ImmutableList.Builder<String> partitionKeysBuilder = ImmutableList.builder();
```
Comment (RussellSpitzer) [inline flink/v1.13/flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java:492]: same comments as above

File: hive3/src/main/java/org/apache/iceberg/mr/hive/serde/objectinspector/IcebergTimestampObjectInspectorHive3.java [hunk 16]
```diff
@@ -50,6 +50,7 @@ public LocalDateTime convert(Object o) {
   }
 
   @Override
+  @SuppressWarnings("JavaLocalDateTimeGetNano")
```
Comment (RussellSpitzer) [inline hive3/src/main/java/org/apache/iceberg/mr/hive/serde/objectinspector/IcebergTimestampObjectInspectorHive3.java:53]: Why is this ok to surpress, it does look like we are doing this incorrectly here, shouldn't this be Timestamp.valueOf(LocalDateTime time)?

I would separate this out into another PR since we could be doing a semantic change here

**Review by RussellSpitzer — CHANGES_REQUESTED** [review]
Most of these look fine, I would if possible (and I know I asked to group these together) separate out the changes which make semantic changes to the code into their own PR's.

For example

"Removing an initialization that sets a variable to null" Good, keep here doesn't actually change anything
"Removing a fully defined type that is already imported" good keep here
"Supressions we know are false positive" Keep here
"Removing empty java docs, fixing java docs" Keep here
"Changing parens" Keep here

--

"Changes an iterator to a list" Separate PR
"Changing the access modifiers on all the JDBC Static Values" Separate PR
"Changes in date handling classes" Separate PR

File: arrow/src/main/java/org/apache/iceberg/arrow/vectorized/ArrowReader.java (hunk 10 above)
Comment (felixYyu) [reply arrow/src/main/java/org/apache/iceberg/arrow/vectorized/ArrowReader.java:271]: revert this

File: core/src/main/java/org/apache/iceberg/BaseFilesTable.java [hunk 17]
```diff
@@ -108,7 +108,8 @@ public TableScan appendsAfter(long fromSnapshotId) {
     }
 
     /**
-     * @return list of manifest files to explore for this files metadata table scan
+     * Returns list of manifest files to explore for this file's metadata table scan
```
[the remark this answers is not in the document]
Comment (felixYyu) [reply core/src/main/java/org/apache/iceberg/BaseFilesTable.java:111]: thanks，I got it and fix.

File: core/src/main/java/org/apache/iceberg/jdbc/JdbcUtil.java [hunk 18]
```diff
@@ -32,20 +32,20 @@
 
 final class JdbcUtil {
   // Catalog Table
-  protected static final String CATALOG_TABLE_NAME = "iceberg_tables";
-  protected static final String CATALOG_NAME = "catalog_name";
-  protected static final String TABLE_NAMESPACE = "table_namespace";
-  protected static final String TABLE_NAME = "table_name";
-  protected static final String METADATA_LOCATION = "metadata_location";
-  protected static final String PREVIOUS_METADATA_LOCATION = "previous_metadata_location";
+  public static final String CATALOG_TABLE_NAME = "iceberg_tables";
```
[the remark this answers is not in the document]
Comment (felixYyu) [reply core/src/main/java/org/apache/iceberg/jdbc/JdbcUtil.java:35]: both JdbcCatalog and JdbcTableOperations need the var of JdbcUtil, so I think the vars of JdbcUtil need to be public.

File: core/src/main/java/org/apache/iceberg/util/Tasks.java (hunk 11 above)
Comment (felixYyu) [reply core/src/main/java/org/apache/iceberg/util/Tasks.java:569]: as ArrowReader，revert this.

File: flink/v1.12/flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java (hunk 14 above)
[the remark this answers is not in the document]
Comment (felixYyu) [reply flink/v1.12/flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java:492]: fixed

File: flink/v1.12/flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java (hunk 12 above)
Comment (felixYyu) [reply flink/v1.12/flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java:498]: fixed

File: flink/v1.12/flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java (hunk 13 above)
Comment (felixYyu) [reply flink/v1.12/flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java:504]: fixed

File: flink/v1.13/flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java (hunk 15 above)
Comment (felixYyu) [reply flink/v1.13/flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java:492]: fixed

File: hive3/src/main/java/org/apache/iceberg/mr/hive/serde/objectinspector/IcebergTimestampObjectInspectorHive3.java (hunk 16 above)
Comment (felixYyu) [reply hive3/src/main/java/org/apache/iceberg/mr/hive/serde/objectinspector/IcebergTimestampObjectInspectorHive3.java:53]: fixed with Timestamp.valueOf(LocalDateTime time)

**Comment by felixYyu** [comment]
I will separate out the changes which make semantic changes to the code into their own PR's.

**Comment by felixYyu** [comment]
The task was not fully completed，don't turn it off. It will be finished today. @RussellSpitzer
